scality/metalk8s
# Run all tests in multi-nodes deployment in CI

## What

This PR introduces a new VM in the Terraform multi-nodes spawn process, called _**bastion**_. It allows one to have access to the private networks spawned, unlike the Eve worker.

The post-install tests (defined under `tests/post`) are now all run without custom options in the multi-nodes deployment context (previously, we invoked `pytest` with custom CLI options `--bootstrap-ip` and `--skip-tls-verify` for performing KubeAPI calls from outside the control-plane network).

## How to test

Look at Eve build reports, see that the tests all run in the multi-nodes stage (no test is skipped).

## Left to do

Some tasks, related to this PR, will need to be tackled in (yet-to-be-created) other issues:

- One would want the bastion to run the install scenario instead of the worker
- One would want to have a more friendly Terraform tooling, usable locally
- One would like to remove `kubectl` invocations in tests, and only use the Python Kubernetes client instead
